### PR TITLE
Add TapAction For ActionSheetPickerView And Update The ActionSheetPickerView frame when toolbar hiden

### LINKED ISSUE #80: Dismiss action sheet by tapping on background
The standard action sheet (UIActionSheet) has a behaviour whereby if you tap on the background (which is not the action sheet), it dismisses the sheet.  I think the same thing should occur for this library too.  

To implement, I put a tap recogniser on the window in the SWActionSheet class to detect the tap, but that seems a little messy, and was thinking there must be a cleaner way to do it - thoughts?

Thanks
Jerrold

### PR DESCRIPTION
**Pull request by BruceZCQ**
update masterview frame and pickerview frame where toolbar hidden; add tap dismiss action actionsheet's window. reflect didSucceedWithAction when tap action fired.

This merge will fix #80.

### CODE REVIEW

File: Pickers/AbstractActionSheetPicker.m [hunk 1]
```diff
@@ -101,24 +100,24 @@ - (instancetype)init
...
         self.popoverBackgroundViewClass = nil;
-
+        
         self.supportedInterfaceOrientations = (UIInterfaceOrientationMask) [[UIApplication sharedApplication]
-                                                                                           supportedInterfaceOrientationsForWindow:
-                                                                                                   [UIApplication sharedApplication].keyWindow];
-
+                                                                            supportedInterfaceOrientationsForWindow:
+                                                                            [UIApplication sharedApplication].keyWindow];
+
```
Comment (skywinder) [inline Pickers/AbstractActionSheetPicker.m]: @BruceZCQ you still have unused trailing whitespaces in empty lines (in whole file). You have to remove them all.

**Comment by BruceZCQ** [comment]
@skywinder , I removed unused trailing whitespaces in empty lines.

File: Pickers/AbstractActionSheetPicker.m [hunk 2]
```diff
@@ -101,20 +100,15 @@ - (instancetype)init
...
+                                                                            [UIApplication sharedApplication].keyWindow];
         UIBarButtonItem *sysDoneButton = [self createButtonWithType:UIBarButtonSystemItemDone target:self
                                                              action:@selector(actionPickerDone:)];
-
         UIBarButtonItem *sysCancelButton = [self createButtonWithType:UIBarButtonSystemItemCancel target:self
                                                                action:@selector(actionPickerCancel:)];
-
         [self setCancelBarButtonItem:sysCancelButton];
         [self setDoneBarButtonItem:sysDoneButton];
-
```
Comment (skywinder) [inline Pickers/AbstractActionSheetPicker.m]: Thanks, almost done.
But -now you actually delete whole string, not only whitespace on it (e.g .\n symbol). Please, try to put it back. You can look here differences: https://github.com/skywinder/ActionSheetPicker-3.0/pull/140/files And here is a red lines. Please, remove only whitespaces, not empty strings. I think the problem in some auto-format settings in your IDE.

**Comment by BruceZCQ** [comment]
@skywinder , sorry the format updated. thanks for your help and patience.

**Comment by BruceZCQ** [comment]
@skywinder , why when I use diff tool the file not change more, push to github the file change more?

**Comment by skywinder** [comment]
@BruceZCQ which diff tool are you use? I believe there is should be some setting option such "ignore string, that contain only white-spaces".

**Comment by BruceZCQ** [comment]
@skywinder configed git like 
```
git config core.autocrlf true
git config core.safecrlf true
```
and pushed the code.

**Comment by skywinder** [comment]
@BruceZCQ :congratulations: You did it!! Thanks!  :+1:

**Comment by skywinder** [comment]
I tested it: everything is good, but this it not obvious, which action should be executed: `cancel` or `success` action by pressing outside.
I will update your request and to support enable/disable option for dismiss picker by tap outside and then merge your code.

**Comment by BruceZCQ** [comment]
@skywinder , Thank you!

**Comment by skywinder** [comment]
This merge will fix #80.

**Pull request merged**